src/components/views/rooms/MemberList.tsx

```diff
@@ -48,199 +30,39 @@ import { ButtonEvent } from "../elements/AccessibleButton";
-        this.state = this.getMembersState([], []);
-        this.showPresence = context?.memberListStore.isPresenceEnabled() ?? true;
-        this.mounted = true;
-        this.listenForMembersChanges();
-    }
-
-    private listenForMembersChanges(): void {
-        const cli = MatrixClientPeg.safeGet();
-        cli.on(RoomStateEvent.Update, this.onRoomStateUpdate);
-        cli.on(RoomMemberEvent.Name, this.onRoomMemberName);
-        cli.on(RoomStateEvent.Events, this.onRoomStateEvent);
-        // We listen for changes to the lastPresenceTs which is essentially
-        // listening for all presence events (we display most of not all of
-        // the information contained in presence events).
-        cli.on(UserEvent.LastPresenceTs, this.onUserPresenceChange);
-        cli.on(UserEvent.Presence, this.onUserPresenceChange);
-        cli.on(UserEvent.CurrentlyActive, this.onUserPresenceChange);
-        cli.on(ClientEvent.Room, this.onRoom); // invites & joining after peek
-        cli.on(RoomEvent.MyMembership, this.onMyMembership);
-    }
-
-    public componentDidMount(): void {
-        this.updateListNow(true);
-    }
-
-    public componentWillUnmount(): void {
-        this.mounted = false;
-        const cli = MatrixClientPeg.get();
-        if (cli) {
-            cli.removeListener(RoomStateEvent.Update, this.onRoomStateUpdate);
-            cli.removeListener(RoomMemberEvent.Name, this.onRoomMemberName);
-            cli.removeListener(RoomEvent.MyMembership, this.onMyMembership);
-            cli.removeListener(RoomStateEvent.Events, this.onRoomStateEvent);
-            cli.removeListener(ClientEvent.Room, this.onRoom);
-            cli.removeListener(UserEvent.LastPresenceTs, this.onUserPresenceChange);
-            cli.removeListener(UserEvent.Presence, this.onUserPresenceChange);
-            cli.removeListener(UserEvent.CurrentlyActive, this.onUserPresenceChange);
-        }
-
-        // cancel any pending calls to the rate_limited_funcs
-        this.updateList.cancel();
-    }
-
-    private get canInvite(): boolean {
-        const cli = MatrixClientPeg.safeGet();
-        const room = cli.getRoom(this.props.roomId);
-
-        return !!room && canInviteTo(room);
-    }
-
-    private getMembersState(invitedMembers: Array<RoomMember>, joinedMembers: Array<RoomMember>): IState {
-        return {
-            loading: false,
-            filteredJoinedMembers: joinedMembers,
-            filteredInvitedMembers: invitedMembers,
-            canInvite: this.canInvite,
-
-            // ideally we'd size this to the page height, but
-            // in practice I find that a little constraining
-            truncateAtJoined: INITIAL_LOAD_NUM_MEMBERS,
-            truncateAtInvited: INITIAL_LOAD_NUM_INVITED,
+    let viewModel = useMemberListViewModel(propsIn.roomId)
+    let props = {...propsIn, ...viewModel}
+    useEffect(() => {
+        viewModel.load()
+        return () => { 
+            viewModel.unload()
```

Comment:
Why can't this load/unload be done by the `useMemberListViewModel` hook? or at least have a `useModel(viewModel)` hook.